### ISSUE TITLE
Ability to manually refresh app inbox

### ISSUE THREAD

**Issue by spideythewebhead** [CleverTap/clevertap-flutter#252]
Hello,

Currently there is no ability to refresh the app inbox - let's say then app is resumed.
And the currently behaviour seems to be that the inbox is refreshed only when launching the app from a killed state.

**Comment by spideythewebhead** [comment]
@Aditi3 @Anush-Shand ^^

**Comment by Anush-Shand** [comment]
@spideythewebhead
Could you elaborate on what do you mean by refresh and what is your use-case?
Currently, everytime a new app inbox message is received `inboxMessagesDidUpdate()` callback is triggered which can then be used to refresh the inbox. Please refer to the below doc for the same
https://developer.clevertap.com/docs/flutter-app-inbox#flutter

**Comment by spideythewebhead** [comment]
@Anush-Shand `inboxMessagesDidUpdate` does not trigger if there is a new inbox message but only the first time the app is launched. 

I have used both types of campaigns

1. `Push notification campaign with copying the message into app inbox` 

    In this case, even tho I receive the push notification instantly, the inbox doesn't receive in realtime the message. I have to kill the app and start it again (which it could take from 5 to 20 minutes doing the same thing before it appears)
    
1. `App inbox campaign`

---

The use case of being able to refresh the inbox, would be because it doesn't work in realtime when a campaign is executed.

**Comment by Anush-Shand** [comment]
@spideythewebhead 

- I rechecked on the sample app from this repo and `inboxMessagesDidUpdate` is triggered everytime a new inbox message is received from any kind of a campaign.
- Could you please share the verbose logs from when a particular event triggers an inbox campaign for us to further debug this.
- Could you also please refer to the [docs](https://developer.clevertap.com/docs/flutter-app-inbox#flutter) and verify your integration.

**Comment by spideythewebhead** [comment]
Hello, I will try using the example project with my credentials and see how it behaves

**Comment by spideythewebhead** [comment]
Hello @Anush-Shand,

After some trials I have noticed the following behavior:

If I create a campaign with `Qualification Criteria` to be `Live behavior` then the app inbox is updated in realtime when an event is triggered. 
If I use `Past behavior/Custom List` (See screenshot) then this doesn't get updated in realtime when the campaign is finished but rather randomly.

I have replicated the behavior in the example project + my own project. I have the same results as describe above in both apps.

![image](https://github.com/user-attachments/assets/c2d56143-3926-4366-b6d7-69dadd7713eb)
